lr_listener.go

```diff
@@ -218,6 +221,19 @@ func (l *LogicalReplicationListener) processMessage(msg *pgx.ReplicationMessage)
 	if err != nil {
 		l.logger.WithError(err).Error("failed to parse wal2json message")
 		l.errCh <- fmt.Errorf("failed to parse wal2json: %v", err)
+		return
+	}
+
+	lsn, err := pgx.ParseLSN(w2jmsg.NextLSN)
+	if err != nil {
+		l.logger.WithError(err).Error("failed to parse LSN from WAL message")
+		l.errCh <- fmt.Errorf("failed to parse LSN from wal2json message: %v", err)
+		return
+	}
+	if lsn > l.replLSN {
+		l.replLSN = lsn
+	} else {
+		l.logger.Warnf("received WAL with LSN %v, which is less than the current LSN %v", lsn, l.replLSN)
```

Comment:
what if lsn == l.replLSN? idk if that's an error or not but certainly the log message is wrong in that case

Reply:
good catch, 37f611718d5d9b56eb2aa944b228c63647a8c813